### PR TITLE
Updates React page to pull data from sanity

### PR DESCRIPTION
**Pull request by zacjones93**
In-Depth Up-to-Date React Tutorials for 2021  egghead io

In-Depth Up-to-Date React Tutorials for 2021  egghead io

hot dog

### CODE REVIEW

File: src/components/search/curated/react/index.tsx [hunk 1]
```diff
@@ -200,4 +201,27 @@ You can find courses below curated just for you whether you're looking for a par
...
+       image,
+       'background': images[label == 'feature-card-background'][0].url,
+       'instructor': collaborators[]->[role == 'instructor'][0]{
+         'name': person->.name
+       },
+     }
+    }
+	},
+ }
+`
```
Comment (zacjones93) [inline src/components/search/curated/react/index.tsx:225]: export query from curated react page

File: src/components/search/curated/react/index.tsx [hunk 2]
```diff
@@ -31,13 +32,13 @@ const SearchReact = () => {
...
   })
-  const stateManagementQuickly: any = find(reactPageData, {
-    id: 'state-management-quickly',
-  })
   const reactArticles: any = find(reactPageData, {id: 'articles'})
   const reactTalks: any = find(reactPageData, {id: 'talks'})
   const reactPodcasts: any = find(reactPageData, {id: 'podcasts'})
 
+  const {reactStateManagement} = topic
+  const {recoilCollection} = reactStateManagement
```
Comment (zacjones93) [inline src/components/search/curated/react/index.tsx:40]: destructure data that's pulled in from sanity

File: src/lib/tags.ts [hunk 3]
```diff
@@ -42,5 +44,31 @@ export async function loadTag(slug: string) {
...
+}
+
+export const loadSanityTag = async (selectedTag: string) => {
+  if (!canLoadSanityTag(selectedTag)) return
+
+  const query = sanityTagPageHash[selectedTag]
+  if (!query) return
+
+  return await sanityClient.fetch(query)
 }
```
Comment (zacjones93) [inline src/lib/tags.ts:74]: load sanity data if query is present

File: src/lib/tags.ts [hunk 4]
```diff
@@ -42,5 +44,31 @@ export async function loadTag(slug: string) {
...
+  react: reactPageQuery,
+}
+
+type SelectedTag = keyof typeof sanityTagPageHash
+
+const canLoadSanityTag = (selectedTag: string): selectedTag is SelectedTag => {
+  const keyNames = Object.keys(sanityTagPageHash)
+
+  return keyNames.includes(selectedTag)
+}
```
Comment (zacjones93) [inline src/lib/tags.ts:65]: set react query for use in `canLoadSanityTag` guard

File: src/lib/tags.ts [hunk 5]
```diff
@@ -42,5 +44,31 @@ export async function loadTag(slug: string) {
...
   const {tag} = await graphQLClient.request(query, variables)
 
-  return tag
+  let sanityTag
+  if (canLoadSanityTag(slug)) {
+    sanityTag = await loadSanityTag(slug)
+  }
+
+  return {...tag, ...sanityTag}
+}
```
Comment (zacjones93) [inline src/lib/tags.ts:53]: merge sanity data and egghead-rails data

File: src/pages/q/[[...all]].tsx [hunk 6]
```diff
@@ -215,13 +215,6 @@ export const getServerSideProps: GetServerSideProps = async function ({
       getInstructorSlugFromInstructorList(selectedInstructors)
     try {
       initialInstructor = await loadInstructor(instructorSlug)
-
-      sanityInstructor = await loadSanityInstructor(instructorSlug)
-
-      initialInstructor = {
-        ...initialInstructor,
-        ...sanityInstructor,
-      }
```
Comment (zacjones93) [inline src/pages/q/[[...all]].tsx:224]: remove extra call to sanity..

**Review by joelhooks — APPROVED** [review]
I'd love to see this evolve into a stage management "feature page" similar to how we made digital gardening, so it is it's own landing page AND usable data for features like this.

**Pull request merged**